### ISSUE TITLE
Could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error

### ISSUE THREAD

**Issue by CyQ239** [google-research/football#104]
When I set extra_players=['ppo2_cnn:right_players=1,policy=gfootball_impala_cnn,checkpoint=./01600']
I get the error"could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error", and the program hang out
but when I set extra_players=None, it works fine
how can I handle this problem?
Many thanks

**Comment by AntonRaichuk** [comment]
Could you provide more logs / stacktrace of the failure?

Also maybe explain a bit your setup? Do you have GPU? Was the checkpoint trained on the same machine?

**Comment by CyQ239** [comment]
Thank you for yout reply!

gfootball2.0.3
extra_players=['ppo2_cnn:right_players=1,policy=gfootball_impala_cnn,checkpoint=./01600'] in "/gfootball/env/__init__.py" 
run program by using "python3 -m gfootball.examples.run_ppo2 --level=11_vs_11_hard_stochastic --policy=gfootball_impala_cnn --num_envs=16 --nsteps=512 --noptepochs=2 --nminibatches=4 --lr=0.00011879 --ent_coef=0.00155 --gamma=0.997 --max_grad_norm=0.76 --cliprange=0.115  --render=False --load_path=./01600"

program logs are as followed:
/home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:526: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:527: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:528: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:529: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:530: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:535: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.
Logging to ./output
2019-12-13 09:36:23.003283: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-12-13 09:36:23.399380: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-13 09:36:23.407691: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:998] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-12-13 09:36:23.424141: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x1e7de70 executing computations on platform CUDA. Devices:
2019-12-13 09:36:23.424156: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): GeForce GTX 1080 Ti, Compute Capability 6.1
2019-12-13 09:36:23.424161: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (1): GeForce GTX 1080 Ti, Compute Capability 6.1
2019-12-13 09:36:23.568525: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3696000000 Hz
2019-12-13 09:36:23.570676: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x1b0f430 executing computations on platform Host. Devices:
2019-12-13 09:36:23.570743: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2019-12-13 09:36:23.571046: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.645
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 10.43GiB
2019-12-13 09:36:23.571169: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 1 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.645
pciBusID: 0000:09:00.0
totalMemory: 10.92GiB freeMemory: 10.77GiB
2019-12-13 09:36:23.575541: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0, 1
2019-12-13 09:36:23.605994: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-13 09:36:23.606063: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 1 
2019-12-13 09:36:23.606088: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N Y 
2019-12-13 09:36:23.606112: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 1:   Y N 
2019-12-13 09:36:23.606292: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10145 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
2019-12-13 09:36:23.606824: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 10479 MB memory) -> physical GPU (device: 1, name: GeForce GTX 1080 Ti, pci bus id: 0000:09:00.0, compute capability: 6.1)
2019-12-13 09:36:23.878828: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.878824: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879004: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.878841: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879335: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879349: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879363: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879409: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879554: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879878: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879848: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.879875: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.880171: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.880209: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.880508: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.880834: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.880850: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.881113: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.881321: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.881345: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.881632: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.881679: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.878851: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.878854: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.878854: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.884529: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.878827: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.884529: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.884593: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.888664: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.889739: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
2019-12-13 09:36:23.890002: E tensorflow/stream_executor/cuda/cuda_driver.cc:1204] could not retrieve CUDA device count: CUDA_ERROR_NOT_INITIALIZED: initialization error
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/input.py:57: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.

WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
WARNING:tensorflow:From /home/cyq/.local/lib/python3.6/site-packages/baselines/common/policies.py:43: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.


then my program hangs out and I have to kill my program to finish
Driver Version: 430.50 CUDA10.0 
my pip list:
absl-py (0.8.1)
apturl (0.5.2)
asn1crypto (0.24.0)
astor (0.8.0)
baselines (0.1.6)
Brlapi (0.6.6)
cachetools (3.1.1)
certifi (2019.11.28)
chardet (3.0.4)
Click (7.0)
cloudpickle (1.2.2)
command-not-found (0.3)
contextlib2 (0.6.0.post1)
cryptography (2.1.4)
cupshelpers (1.0)
cycler (0.10.0)
decorator (4.4.1)
defer (1.0.6)
distro-info (0.18ubuntu0.18.04.1)
dm-sonnet (1.23)
fire (0.2.1)
future (0.18.2)
gast (0.3.2)
gfootball (2.0.3)
google-api-core (1.14.3)
google-auth (1.7.1)
google-cloud-core (1.0.3)
google-cloud-storage (1.23.0)
google-pasta (0.1.8)
google-resumable-media (0.5.0)
googleapis-common-protos (1.6.0)
grpcio (1.25.0)
gurobipy (8.1.0)
gym (0.15.4)
h5py (2.10.0)
httplib2 (0.9.2)
idna (2.8)
joblib (0.14.0)
Keras-Applications (1.0.8)
Keras-Preprocessing (1.1.0)
keyring (10.6.0)
keyrings.alt (3.0)
kiwisolver (1.1.0)
language-selector (0.1)
launchpadlib (1.10.6)
lazr.restfulclient (0.13.5)
lazr.uri (1.0.3)
louis (3.5.0)
macaroonbakery (1.1.3)
Mako (1.0.7)
Markdown (3.1.1)
MarkupSafe (1.0)
matplotlib (3.1.1)
mock (3.0.5)
netifaces (0.10.4)
numpy (1.17.4)
oauth (1.0.1)
olefile (0.45.1)
opencv-python (4.1.2.30)
pexpect (4.2.1)
Pillow (6.2.0)
pip (9.0.1)
progressbar (2.5)
protobuf (3.11.1)
pyasn1 (0.4.8)
pyasn1-modules (0.2.7)
pycairo (1.16.2)
pycrypto (2.6.1)
pycups (1.9.73)
pygame (1.9.6)
pyglet (1.3.2)
pygobject (3.26.1)
pymacaroons (0.13.0)
PyNaCl (1.1.2)
pyparsing (2.4.2)
pyRFC3339 (1.0)
python-apt (1.6.4)
python-dateutil (2.8.0)
python-debian (0.1.32)
pytz (2019.3)
pyxdg (0.25)
PyYAML (3.12)
reportlab (3.4.0)
requests (2.22.0)
requests-unixsocket (0.1.5)
rsa (4.0)
scipy (1.3.3)
screen-resolution-extra (0.0.0)
SecretStorage (2.3.1)
semantic-version (2.8.3)
setuptools (42.0.2)
simplejson (3.13.2)
six (1.13.0)
system-service (0.3)
systemd-python (234)
tensorboard (1.13.1)
tensorflow-estimator (1.13.0)
tensorflow-gpu (1.13.1)
tensorflow-probability (0.8.0)
termcolor (1.1.0)
torch (1.3.0)
torchvision (0.4.1)
tqdm (4.36.1)
ubuntu-drivers-common (0.0.0)
ufw (0.36)
unattended-upgrades (0.1)
urllib3 (1.25.7)
usb-creator (0.3.3)
wadllib (1.3.2)
Werkzeug (0.16.0)
wheel (0.33.6)
wrapt (1.11.2)
xkit (0.0.0)
zope.interface (4.3.2)

**Comment by AntonRaichuk** [comment]
When I did the same steps as you, I did not get a CUDA_ERROR_NOT_INITIALIZED error, but my run did get stuck without making any progress.
It seems that the problem is with tensorflow not being fork safe.

what worked for me is the following hack (all changes are in run_ppo2.py):
1) replace "import tensorflow as tf"  with
"from absl import app
from absl import flags"
2) replace "flags = tf.app.flags
FLAGS = tf.app.flags.FLAGS"
with  "FLAGS = flags.FLAGS"
3) replace "train()" with "app.run(train)"
4) replace "def train():" with "def train(_):"
5) move "  vec_env = SubprocVecEnv([
      (lambda _i=i: create_single_football_env(_i))
      for i in range(FLAGS.num_envs)
  ], context=None)" to the beginning of "train" function
6) add "import tensorflow as tf" right after


Basically it just postpones import of tensorflow after subprocesses are created.
This worked for me.

Could you try if that works for you, and if does, maybe we'll merge this change to the master branch.

**Comment by CyQ239** [comment]
It does work for me!
Thank you very much!

I also replace "self._sess = tf.Session()" with 
"config = tf.ConfigProto()
  config.gpu_options.allow_growth = True
  tf.Session(config=config)"
in gfootball/env/players/ppo2_cnn.py
otherwise I'll get a "cuda out of memory" error

**Comment by AntonRaichuk** [comment]
Thanks for the additional fix!

We'll make sure to include it in the next release
